samples/drivers/gecko_adc/src/bq24250.c

```diff
@@ -0,0 +1,330 @@
+const struct device *gpiof_dev = DEVICE_DT_GET(DT_NODELABEL(gpiof));
+
+static const struct device *i2c1_dev = DEVICE_DT_GET(DT_NODELABEL(i2c1));
+
+/* BQ Battery Charger interrupt callback */
+static struct gpio_callback gpio_batt_chrg_cb;
+int bq_battery_charge_int_isr_count;
+bool bq_battery_charge_int_state_change;
+uint8_t i2c_read_data[20];
+
+static int i2c_read_to_buffer(uint8_t dev_addr, int reg_addr, uint8_t *buf, uint8_t buf_length)
+{
+	const struct device *dev;
+	uint8_t reg_addr_buf[MAX_BYTES_FOR_REGISTER_INDEX];
+	int reg_addr_bytes;
+	int ret;
+
+	dev = i2c1_dev;
+	if (!dev) {
+		printf("I2C: Device driver I2C_1 not found.");
+		return -ENODEV;
+	}
+
+	reg_addr_bytes = 1;
+	sys_put_be32(reg_addr, reg_addr_buf);
+
+	ret = i2c_write_read(dev, dev_addr,
+			reg_addr_buf +
+			MAX_BYTES_FOR_REGISTER_INDEX - reg_addr_bytes,
+			reg_addr_bytes, buf, buf_length);
+	if (ret < 0) {
+		printf("Failed to read from device: 0x%x", dev_addr);
+		return -EIO;
+	}
+	return 0;
+}
+
+int get_bq24250_status(uint8_t *charging, uint8_t *vbus, uint8_t *attached, uint8_t *fault)
+{
+	int ret = 0;
+	uint8_t bq_addr = 0x6a;
+	uint8_t rx_len = 1;
+	uint8_t bq_reg_0x00 = 0x00;
+	uint8_t bq_reg_0x00_data;
+
+	ret = i2c_read_to_buffer(bq_addr, bq_reg_0x00, &bq_reg_0x00_data, rx_len);
+	if (ret) {
+		printf("read_bytes function:Error reading from BQ! error code (%d)\n", ret);
+	}
+
+	/* parse the BQ Charge Status or Fault Interrupt that got us here. */
+	uint8_t charger_status = ((bq_reg_0x00_data & 0x30) >> 4);
+	uint8_t charge_fault_status = (bq_reg_0x00_data & 0x0F);
+
+	*vbus = 0;
+	*charging = 0;
+	*attached = 0;
+	*fault = 0;
+
+	if ((charger_status == 3) && (charge_fault_status > 0)) {
+		*vbus = 0;
+		*charging = 0;
+		if (charge_fault_status == 8) {
+			*attached = 0;
+		} else {
+			*attached = 1;
+			*fault = 1;
+		}
+	} else {
+		if ((charger_status == 1) && (charge_fault_status == 0)) {
+			*vbus = 1;
+			*charging = 1;
+			*attached = 1;
+		} else {
+			*vbus = 1;
+			*charging = 0;
+			*attached = 1;
+		}
+	}
+	return ret;
+}
+
+static int check_battery_charger_regs(void)
+{
+	int ret = 0;
+
+	static bool no_vbus = true;
+
+	uint8_t bq_addr = 0x6a;
+	uint8_t rx_len = 1;
+
+	uint8_t bq_reg_0x00 = 0x00;
+	uint8_t bq_reg_0x00_data = 0;
+
+	uint8_t bq_reg_0x01 = 0x01;
+	uint8_t bq_reg_0x01_data = 0;
+
+	ret = i2c_read_to_buffer(bq_addr, bq_reg_0x00, &bq_reg_0x00_data, rx_len);
+	if (ret) {
+		printf("i2c read_bytes function:Error reading from BQ! error code (%d)\n", ret);
+	}
+
+	ret = i2c_read_to_buffer(bq_addr, bq_reg_0x01, &bq_reg_0x01_data, rx_len);
+	if (ret) {
+		printf("i2c read_bytes function:Error reading from BQ! error code (%d)\n", ret);
+	}
+
+	k_msleep(100);
+
+	ret = i2c_read_to_buffer(bq_addr, bq_reg_0x00, &bq_reg_0x00_data, rx_len);
+	if (ret) {
+		printf("i2c read_bytes function:Error reading from BQ! error code (%d)\n", ret);
+	}
+
+	ret = i2c_read_to_buffer(bq_addr, bq_reg_0x01, &bq_reg_0x01_data, rx_len);
```

Comment:
rx_len should be replaced with the following:
`sizeof(bq_reg_0x01_data)`

Reply:
Done